pkg/receiver/activedirectoryinvreceiver/adinvreceiver.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package activedirectoryinvreceiver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	adsi "github.com/go-adsi/adsi"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+// Client is an interface for an Active Directory client
+type Client interface {
+	Open(path string, resourceLogs *plog.ResourceLogs) (Container, error)
+}
+
+// ADSIClient is a wrapper for an Active Directory client
+type ADSIClient struct{}
+
+// Open an Active Directory container
+func (c *ADSIClient) Open(path string, resourceLogs *plog.ResourceLogs) (Container, error) {
+	client, err := adsi.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	ldapPath := fmt.Sprintf("LDAP://%s", path)
+	root, err := client.Open(ldapPath)
+	if err != nil {
+		return nil, err
+	}
+	rootContainer, err := root.ToContainer()
+	if err != nil {
+		return nil, err
+	}
+	windowsContainer := &ADSIContainer{rootContainer}
+	return windowsContainer, nil
+}
+
+type ADReceiver struct {
+	config   *ADConfig
+	logger   *zap.Logger
+	client   Client
+	consumer consumer.Logs
+	wg       *sync.WaitGroup
+	doneChan chan bool
+}
+
+// newLogsReceiver creates a new Active Directory Inventory receiver
+func newLogsReceiver(cfg *ADConfig, logger *zap.Logger, client Client, consumer consumer.Logs) *ADReceiver {
+
+	return &ADReceiver{
+		config:   cfg,
+		logger:   logger,
+		client:   client,
+		consumer: consumer,
+		wg:       &sync.WaitGroup{},
+		doneChan: make(chan bool),
+	}
+}
+
+// Start the logs receiver
+func (l *ADReceiver) Start(ctx context.Context, _ component.Host) error {
+	l.logger.Debug("Starting to poll for active directory inventory records")
+	l.wg.Add(1)
+	go l.startPolling(ctx)
+	return nil
+}
+
+// Shutdown the logs receiver
+func (l *ADReceiver) Shutdown(_ context.Context) error {
+	l.logger.Debug("Shutting down logs receiver")
+	close(l.doneChan)
+	l.wg.Wait()
+	return nil
+}
+
+// Start polling for Active Directory inventory records
+func (l *ADReceiver) startPolling(ctx context.Context) {
+	defer l.wg.Done()
+	t := time.NewTicker(l.config.PollInterval * time.Second)
```

Comment:
This multiplying of the `PollInterval` by `time.Second` is misleading. The `PollInterval` is already a duration, so treat it like one, instead of treating it like the number of seconds.

Reply:
changed the poll interval to accept a duration string like "24h".